pathsaver/src/main/kotlin/PathsaverPlugin.kt

```diff
@@ -0,0 +1,101 @@
+package kotlinx.knit.pathsaver
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import org.jetbrains.dokka.CoreExtensions
+import org.jetbrains.dokka.base.DokkaBase
+import org.jetbrains.dokka.base.resolvers.local.LocationProvider
+import org.jetbrains.dokka.links.DRI
+import org.jetbrains.dokka.model.DModule
+import org.jetbrains.dokka.model.DisplaySourceSet
+import org.jetbrains.dokka.model.doc.DocumentationLink
+import org.jetbrains.dokka.model.withDescendants
+import org.jetbrains.dokka.pages.RendererSpecificResourcePage
+import org.jetbrains.dokka.pages.RenderingStrategy
+import org.jetbrains.dokka.pages.RootPageNode
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.plugability.DokkaPlugin
+import org.jetbrains.dokka.plugability.query
+import org.jetbrains.dokka.transformers.documentation.DocumentableTransformer
+import org.jetbrains.dokka.transformers.pages.PageTransformer
+
+const val LINK_INDEX_FILE = "paths-index.json"
+
+class PathsaverPlugin : DokkaPlugin() {
+    private val dokkaBasePlugin by lazy { plugin<DokkaBase>() }
+
+    val transformer by extending {
+        CoreExtensions.documentableTransformer with DriExtractor order { after(dokkaBasePlugin.emptyPackagesFilter) }
+    }
+
+    val preprocessor by extending {
+        dokkaBasePlugin.htmlPreprocessors providing { context -> PathSaver(context, dokkaBasePlugin) }
+    }
+}
+
+object DriExtractor : DocumentableTransformer {
```

Comment:
Could you please drop a comment on what this plugin does and why it is necessary?
Also, could you please elaborate on why it has to be a separate artifact instead of being embedded right onto knit?

Reply:
Sure, I'll document it properly. It can be embedded right into knit, but I think semantically it should be a separate artifact, because this is essentially a Dokka plugin that outputs a `json` file alongside the ordinary documentation. It can be used without Knit and can be developed more or less independently of Knit and Dokka. I can embed it into Knit but I think that adding knit both as a Dokka plugin and as a Gradle plugin in one project could be confusing for some people